src/intents.js

```diff
@@ -3,7 +3,7 @@ import {cozyFetchJSON} from './fetch'
 const intentClass = 'coz-intent'
 
 // inject iframe for service in given element
-function injectService (url, element, data) {
+function injectService (url, element, intent, data) {
   const document = element.ownerDocument
   if (!document) throw new Error('Cannot retrieve document object from given element')
```

Comment:
Is there a reason why this validation code is not in the Promise constructor ?

I think you should not mix synchronous code and asynchronous code.